File: ai-service/app/schemas/common/api_response.py
from typing import Any

from pydantic import BaseModel, Field


class ErrorBody(BaseModel):
    code: str
    message: str
    details: list[dict[str, Any]] = Field(default_factory=list)


class SuccessResponse(BaseModel):
    success: bool = True
    message: str = "Success"
    data: Any = Field(default_factory=dict)


class ErrorResponse(BaseModel):
    success: bool = False
    error: ErrorBody

File: backend/src/utils/http/apiResponse.js
export const sendSuccess = (res, { statusCode = 200, message = "Thành công.", data = {}, meta } = {}) => {
  const payload = {
    success: true,
    message,
    data
  };

  if (meta) {
    payload.meta = meta;
  }

  return res.status(statusCode).json(payload);
};

export const sendError = (
  res,
  {
    statusCode = 500,
    code = "INTERNAL_SERVER_ERROR",
    message = "Đã xảy ra lỗi không mong muốn.",
    details = []
  } = {}
) =>
  res.status(statusCode).json({
    success: false,
    error: {
      code,
      message,
      details
    }
  });
